nucypher/blockchain/eth/actors.py

```diff
@@ -656,3 +676,349 @@ def request_evaluation(self, evidence):
+        return self.__accounts
+
+    def __get_accounts(self) -> None:
+        accounts = self.blockchain.client.accounts
+        self.__accounts.extend(accounts)
+
+    def set_funding_account(self, address: str):
+        self.__funding_account = address
+
+    @property
+    def funding_account(self) -> str:
+        if self.__funding_account is NO_FUNDING_ACCOUNT:
+            raise self.NoFundingAccount
+        return self.__funding_account
+
+    @property
+    def funding_tokens(self) -> NU:
+        nunits = self.token_agent.get_balance(address=self.funding_account)
+        return NU.from_nunits(nunits)
+
+    @property
+    def funding_eth(self) -> Decimal:
+        ethers = self.blockchain.client.get_balance(self.funding_account)
+        return ethers
+
+    #
+    # Staking Utilities
+    #
+
+    def read_onchain_stakes(self, account: str = None) -> None:
+        if account:
+            accounts = [account]
+        else:
+            accounts = self.__accounts
+
+        for account in accounts:
+            stakes = list(self.staking_agent.get_all_stakes(staker_address=account))
+            if stakes:
+                staker = Staker(is_me=True, checksum_address=account, blockchain=self.blockchain)
+                self.__stakers[staker] = staker.stakes
+
+    @property
+    def total_stake(self) -> NU:
+        total = sum(staker.locked_tokens() for staker in self.stakers)
+        return total
+
+    @property
+    def stakers(self) -> List[Staker]:
+        return list(self.__stakers.keys())
+
+    @property
+    def stakes(self) -> list:
+        payload = list()
+        for staker in self.__stakers:
+            payload.extend(staker.stakes)
+        return payload
+
+    @property
+    def account_balances(self) -> dict:
+        balances = dict()
+        for account in self.__accounts:
+            funds = {'ETH': self.blockchain.client.get_balance(account),
+                     'NU': self.token_agent.get_balance(account)}
+            balances.update({account: funds})
+        return balances
+
+    @property
+    def staker_balances(self) -> dict:
+        balances = dict()
+        for staker in self.stakers:
+            staker_funds = {'ETH': staker.eth_balance, 'NU': staker.token_balance}
+            balances = {staker.checksum_address: staker_funds}
+        return balances
+
+    def __serialize_stake_info(self) -> list:
+        payload = list()
+        for staker in self.stakers:
+            stake_info = [stake.to_stake_info() for stake in staker.stakes]
+            worker_address = staker.worker_address or NO_WORKER_ASSIGNED
+            staker_funds = {'ETH': int(staker.eth_balance), 'NU': int(staker.token_balance)}
+            staker_payload = {'staker': staker.checksum_address,
+                              'balances': staker_funds,
+                              'worker': worker_address,
+                              'stakes': stake_info}
+            payload.append(staker_payload)
```

Comment:
It could be cleaner to have the `Staker` class return the payload representation based on its own properties, and encompasses all the data you are using above. This could reduce the code here to something like:
```python
payload = list()
for staker in self.stakers:
    payload.append(staker.get_payload())

return payload
```

Reply:
:heavy_check_mark: